helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixManager.java

```diff
@@ -866,6 +858,83 @@ public void disconnect() {
     }
   }
 
+  /**
+   * The callback handler cleanup operations that require an active ZkClient connection.
```

Comment:
Why this require an active zkclient? CallbackHandler are in memory object, is there anything we need to connection to ZK?

Reply:
1. we subscribe, and the subscription requires ZK connection.
2. Some of our Helix callback handler implementations will read ZK even for finalizing.
3. There is no way to prevent any customer-defined handler logic to access ZK for finalizing.